dist/angular-form-messages-bootstrap.js

```diff
@@ -30,40 +29,99 @@ angular.module('angularFormMessagesBootstrap')
 angular.module('angularFormMessagesBootstrap')
 /**
  * This directive shows the 'has-error', 'has-warning', 'has-info' or 'has-success' when there is one message on the field.
- * Where there are multiple messages, no class is added.
+ * Where there are multiple messages and multiple messages are shown, no class is added.
  * When there is no message (the field is valid) and showSuccess is true, show the 'has-success' class.
  */
-  .directive('afFieldState', ["MESSAGE_TYPES", "MessageService", function (
-    MESSAGE_TYPES,
-    MessageService
+  .directive('afFieldState', ["$timeout", "AfMessageService", "MESSAGE_TYPES", function (
+    $timeout,
+    AfMessageService,
+    MESSAGE_TYPES
   ) {
     return {
-      require: ['^afSubmit', '^form'],
-      link: function ($scope, elem, attrs, ctrls) {
+      require: '^afSubmit',
+      controller: angular.noop,
+      link: function ($scope, elem, attrs, afSubmitCtrl) {
+        function typeToFieldStateClass(messageType) {
+          var mapping = {};
+          mapping[MESSAGE_TYPES[0]] = 'has-success';
+          mapping[MESSAGE_TYPES[1]] = 'has-info';
+          mapping[MESSAGE_TYPES[2]] = 'has-warning';
+          mapping[MESSAGE_TYPES[3]] = 'has-error';
+          return mapping[messageType];
+        }
+
+        function alertClassToType(alertClass) {
+          return {
```

Comment:
Ja maar dat doe je ook als je direct een nieuw object teruggeeft. Dit heb ik gedaan omdat je geen variabele keys kunt hebben wanneer je een object maakt. Mooier is wel om het object buiten de functie te zetten, dus dat ga ik doen.